src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    const isHerbivore = victim instanceof Herbivore;
+
+    if (isHerbivore && !victim.hidden) {
+      victim.health -= 50;
+
+      if (victim.health === 0) {
```

Comment:
Roman suggests to do it like this and Nazar suggests to take in the consideration that if we add another method that may cause health to go below zero. So you decide 😄